miner/Miner.hs

```diff
@@ -0,0 +1,285 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE ViewPatterns #-}
+
+-- |
+-- Module: Main
+-- Copyright: Copyright © 2019 Kadena LLC.
+-- License: MIT
+-- Maintainer: Colin Woodbury <colin@kadena.io>
+-- Stability: experimental
+--
+-- The fast "inner loop" of the mining process. Sits idle upon startup until
+-- work is submitted.
+--
+
+module Main ( main ) where
+
+import BasePrelude hiding (app, option)
+import Control.Concurrent.Async (async, race, wait)
+import Control.Concurrent.STM.TMVar
+import Control.Concurrent.STM.TVar (modifyTVar')
+import Control.Error.Util (note)
+import Control.Scheduler (Comp(..), replicateWork, terminateWith, withScheduler)
+import Crypto.Hash.Algorithms (SHA512t_256)
+import Crypto.Hash.IO
+import qualified Data.ByteArray as BA
+import Data.Bytes.Put
+import qualified Data.ByteString as B
+import Data.HashMap.Strict (HashMap)
+import qualified Data.HashMap.Strict as HM
+import qualified Data.Text as T
+import Data.Tuple.Strict (T2(..))
+import Foreign.Marshal.Alloc (allocaBytes)
+import qualified Network.Wai.Handler.Warp as W
+import Options.Applicative
+import Servant.API
+import Servant.Server (Application, Server, serve)
+import Servant.Swagger (toSwagger)
+import Servant.Swagger.UI (SwaggerSchemaUI, swaggerSchemaUIServer)
```

Comment:
I think, we should try to be consistent across all chainweb components. So, we should use swagger ui everywhere or nowhere.

Generally, do we want swagger ui? I am a bit concerned in overloading our tools and adding problematic dependencies. If a service exports just swagger json, one can still use a stand-alone swagger-ui client to get the same functionality.

Reply:
I'll remove it for now.